src/sidebar/components/menu-keyboard-navigation.js

```diff
@@ -8,27 +8,45 @@ function isElementVisible(element) {
   return element.offsetParent !== null;
 }
 
+/**
+ * @template T
+ * @typedef {import("preact/hooks").Ref<T>} Ref
+ */
+
+/**
+ * @typedef MenuKeyboardNavigationProps
+ * @prop {string} [className]
+ * @prop {(e: KeyboardEvent) => any} [closeMenu] - Callback when the menu is closed via keyboard input
+ * @prop {boolean} [visible] - When  true`, sets focus on the first item in the list
+ * @prop {Object} children - Array of nodes which may contain <MenuItems> or any nodes
+ */
+
 /**
  * Helper component used by Menu and MenuItem to facilitate keyboard navigation of a
  * list of <MenuItem> components. This component should not be used directly.
  *
  * Note that `ArrowRight` shall be handled by the parent <MenuItem> directly and
  * all other focus() related navigation is handled here.
+ *
+ * @param {MenuKeyboardNavigationProps} props
  */
 export default function MenuKeyboardNavigation({
   className,
   closeMenu,
   children,
   visible,
 }) {
-  const menuRef = useRef(null);
+  const menuRef = /** @type {Ref<HTMLDivElement|null>} */ (useRef(null));
```

Comment:
@robertknight , the type needs to be div since its being added to a div element. I was not sure if you're example (in slack) was just using the HTMLFormElement as an example or literally everywhere? I'm assuming it was just an example.

Reply:
Hi Kyle, the element type depends on the use. Depending on the use you may just be able to use `HTMLElement`.

Note in my Slack example that the cast is to the `useRef` _argument_ (ie. `useRef(/** @type {HTMLDivElement|null} / (null))`). Then everything else is inferred and you don’t need to import the Ref type. The need for null checks elsewhere is also currently avoided if you do this.

Reply:
@robertknight , I missed that in your example, sorry. That makes more sense. Don't bother with refs at all, just assert the type of thing in the ref and the rest is easy. got it!